app/src/main/java/com/googleresearch/capturesync/CameraActivity.java

```diff
@@ -211,12 +174,58 @@ public void surfaceDestroyed(SurfaceHolder holder) {
                     Log.i(TAG, "destroyed.");
                 }
             };
-    private Surface viewfinderSurface;
-    private PhaseAlignController phaseAlignController;
-    private int numCaptures;
-    private Toast latestToast;
-    private Surface surface;
 
+    public String getLastTimeStamp() {
+        return lastTimeStamp;
+    }
+
+    public ArrayDeque<Long> getLatestFrames() {
+        return latestFrames;
+    }
+
+    public void displayStreamFrame(File streamFrame) {
+        Bitmap streamBitmap = BitmapFactory.decodeFile(streamFrame.getAbsolutePath());
+        runOnUiThread(
+                () -> streamImageView.setImageBitmap(streamBitmap)
```

Comment:
stream is based on image view? in future it will be better to use surface with rendering